### PR TITLE
[Group] Removed duplicate code for rejoining groups

### LINKED ISSUE #14254: [Group] Refactor code used to rejoin existing groups at boot
#### Problem
* Code to rejoin an existing group is doubled between server.cpp and GenericThreadStackManagerImpl_OpenThread_LwIP.cpp

 #### Proposed Solution
* Create a helper function with the copied code and call it in the appropriate locations
* Manufacturers will need to test groups on their own platforms to validate their devices rejoin existing groups. If not, they will need to do the appropriate changes

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* Fixes #14254 

#### Change overview
* Created helper functions that rejoins exisiting groups

#### Testing
* Manual tests with efr32 MG12 and MG24

### CODE REVIEW

**Review by jepenven-silabs — APPROVED** [review]

File: src/app/server/Server.cpp [hunk 1]
```diff
@@ -295,6 +270,38 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
...
+    ChipLogProgress(AppServer, "Joining Multicast groups");
+    CHIP_ERROR err                     = CHIP_NO_ERROR;
+    ConstFabricIterator fabricIterator = mFabrics.cbegin();
+    while (!fabricIterator.IsAtEnd())
+    {
+        const FabricInfo & fabric = *fabricIterator;
+        Credentials::GroupDataProvider::GroupInfo groupInfo;
+
+        Credentials::GroupDataProvider::GroupInfoIterator * iterator = mGroupsProvider.IterateGroupInfo(fabric.GetFabricIndex());
+        while (iterator->Next(groupInfo))
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply src/app/server/Server.cpp]: added the check

File: src/app/server/Server.cpp [hunk 2]
```diff
@@ -295,6 +270,38 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
...
+        {
+            err = mTransports.MulticastGroupJoinLeave(
+                Transport::PeerAddress::Multicast(fabric.GetFabricIndex(), groupInfo.group_id), true);
+            if (err != CHIP_NO_ERROR)
+            {
+                ChipLogError(AppServer, "Error when trying to join Group %" PRIu16 " of fabric index %u", groupInfo.group_id,
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply src/app/server/Server.cpp]: updated log message

**Review by jepenven-silabs — APPROVED** [review]

File: src/app/server/Server.cpp [hunk 3]
```diff
@@ -254,45 +254,52 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
...
-        ConstFabricIterator fabricIterator = mFabrics.cbegin();
-        while (!fabricIterator.IsAtEnd())
-        {
-            const FabricInfo & fabric = *fabricIterator;
-            Credentials::GroupDataProvider::GroupInfo groupInfo;
+        ChipLogError(AppServer, "ERROR setting up transport: %s", ErrorStr(err));
+    }
+    else
+    {
+        ChipLogProgress(AppServer, "Server Listening...");
```
Comment (tcarmelveilleux) [inline src/app/server/Server.cpp:268]: Suggest adding which port it is listening

File: src/app/server/Server.cpp [hunk 4]
```diff
@@ -254,45 +254,52 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
...
 
-            Credentials::GroupDataProvider::GroupInfoIterator * iterator =
-                mGroupsProvider.IterateGroupInfo(fabric.GetFabricIndex());
+        auto * iterator = mGroupsProvider.IterateGroupInfo(fabric.GetFabricIndex());
+        if (iterator)
+        {
             while (iterator->Next(groupInfo))
             {
                 err = mTransports.MulticastGroupJoinLeave(
                     Transport::PeerAddress::Multicast(fabric.GetFabricIndex(), groupInfo.group_id), true);
```
Comment (tcarmelveilleux) [inline src/app/server/Server.cpp:287]: What does the "true" stand for?

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/server/Server.cpp (hunk 4 above)
Comment (mkardous-silabs) [reply src/app/server/Server.cpp:288]: True means the `GroupDataProvider` was able to allocate an `GroupInfoIterator`.
I added a comment explaining the true scenario

File: src/app/server/Server.cpp (hunk 3 above)
Comment (mkardous-silabs) [reply src/app/server/Server.cpp]: done

Comment (mkardous-silabs) [reply src/app/server/Server.cpp:268]: If you change the log, it breaks a cirque test since it expects this exact string

**Pull request merged**